### ISSUE TITLE
Reverse Lexer

### ISSUE THREAD

**Issue by srhickma** [srhickma/padd#182]
Allow the lexing direction to be specified for a CDFA region, which will determine the direction that the input is scanned. This will be particularly useful for #180. The syntax could look something like:
```
cdfa::forward {
 // Scan these rules forward
}

cdfa::reverse {
 // Scan these rules backward
}

cdfa {
 // Default direction is forward
}
```

**Comment by srhickma** [comment]
This could be implemented using #183

**Comment by srhickma** [comment]
This is no longer needed for #180, as there are simpler ways to achieve a similar lexing style, so closing for now.
